provider/kubernetes.go

```diff
@@ -125,9 +125,15 @@ func (provider *Kubernetes) loadIngresses(k8sClient k8s.Client) (*types.Configur
 					}
 				}
 				if len(r.Host) > 0 {
+					rule := "Host:" + r.Host
+
+					if strings.Contains(r.Host, "*") {
+						rule = "HostRegexp:" + strings.Replace(r.Host, "*", "{subdomain:[a-z]+}", 1)
```

Comment:
The subdomain rule you implemented here does not allow for capital letters, numbers, or dashes/underscores. This means that `ww2.example.com` will not have a properly matched rule.